Lib/doctest.py

```diff
@@ -1037,7 +1037,9 @@ def _find(self, tests, obj, name, module, source_lines, globs, seen):
                 if isinstance(val, staticmethod):
                     val = getattr(obj, valname)
                 if isinstance(val, classmethod):
-                    val = getattr(obj, valname).__func__
+                    # Lookup via __dict__ instead of getattr
+                    # in case it is a classmethod property.
+                    val = obj.__dict__[valname].__func__
```

Comment:
It can be simplified.
```suggestion
                if isinstance(val, (staticmethod, classmethod)):
                    val = val.__func__
```